aten/src/ATen/cuda/CUDAGraph.h

```diff
@@ -1,39 +1,73 @@
+#pragma once
+
 #include <ATen/Tensor.h>
 #include <c10/core/Device.h>
+#include <c10/cuda/CUDAGraphsC10Utils.h>
 #include <c10/cuda/CUDAStream.h>
-#include <ATen/CUDAGeneratorImpl.h>
 
 namespace at {
+
+class CUDAGeneratorImpl;
```

Comment:
our linter has a reasonable complaint here that in CUDAGeneratorImpl.h CUDAGeneratorImpl is defined as `struct`, not `class`